sql/parser/builtins.go

```diff
@@ -399,15 +401,46 @@ var builtins = map[string][]builtin{
 		builtin{
 			types:      typeList{},
 			returnType: DummyDate,
+			fn: func(e EvalContext, args DTuple) (Datum, error) {
+				return DDate{Time: e.StmtTimestamp.Truncate(24 * time.Hour)}, nil
+			},
+		},
+	},
+
+	"statement_timestamp": {nowImpl},
+	"current_timestamp":   {nowImpl},
+	"now":                 {nowImpl},
```

Comment:
I don't feel strongly which way we go. It would be good, however, if functions like `current_date()` and `age()` were defined in terms of `nowImpl` to ensure that we're consistent.